src/main/java/slimeknights/tconstruct/plugin/crt/managers/BeheadingManager.java

```diff
@@ -0,0 +1,49 @@
+package slimeknights.tconstruct.plugin.crt.managers;
+
+import com.blamejared.crafttweaker.api.CraftTweakerAPI;
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker.api.item.IItemStack;
+import com.blamejared.crafttweaker.api.managers.IRecipeManager;
+import com.blamejared.crafttweaker.impl.actions.recipes.ActionAddRecipe;
+import com.blamejared.crafttweaker.impl.actions.recipes.ActionRemoveRecipe;
+import com.blamejared.crafttweaker.impl.entity.MCEntityType;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.util.ResourceLocation;
+import org.openzen.zencode.java.ZenCodeType;
+import slimeknights.mantle.recipe.EntityIngredient;
+import slimeknights.mantle.recipe.ItemOutput;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.library.recipe.modifiers.BeheadingRecipe;
+
+@ZenRegister
+@ZenCodeType.Name("mods.tconstruct.Beheading")
+public class BeheadingManager implements IRecipeManager {
+  
+  @ZenCodeType.Method
+  public void addRecipe(String name, MCEntityType ingredient, IItemStack output) {
+    name = fixRecipeName(name);
+    ResourceLocation id = new ResourceLocation("crafttweaker", name);
+    BeheadingRecipe recipe = new BeheadingRecipe(id, EntityIngredient.of(ingredient.getInternal()), ItemOutput.fromStack(output.getInternal()));
+    CraftTweakerAPI.apply(new ActionAddRecipe(this, recipe));
+  }
+  
+  @Override
+  public void removeRecipe(IItemStack output) {
+    throw new IllegalArgumentException("Cannot remove Beheading Recipes by an IItemStack output! Use `removeRecipe(MCEntityType input)` instead!");
```

Comment:
Why can't you remove recipes by an item stack output? They have an item stack output.

Reply:
The only output I see that I can actually use (without having to pass an entity) in is here
https://github.com/jaredlll08/TinkersConstruct/blob/66512ba258bf6b8986ffb05ec8cbcec677ab14e7/src/main/java/slimeknights/tconstruct/library/recipe/modifiers/BeheadingRecipe.java#L54

Based on that javadoc saying it is only used for JEI, I figured I should probably use a more reliable approach to removals.

I'm happy to use it if you think it is fine though

Reply:
Javadoc is because we have two recipes that extend this one, `PlayerBeheadingRecipe` and `SnowGolemBeheadingRecipe`. Both depend on the entity instance to set the proper NBT and alike. However, if one was to remove by `Items.PLAYER_HEAD`, that would properly remove the player beheading recipe, so I think its fine.